### PR TITLE
ADD: [droid] implement Google Breakpad for minidumps

### PR DESCRIPTION
**Pull request by koying**
Implement minidumps for android (no more logcat).

2 sided:
- At build time, symbols are created and must be saved
- At crash time, minidumps are created in our TEMP directory

Reconstruction of the stacktrace is done with:
`minidump_stackwalk <minidimp> <symbols_dir>`

TODO:
- jenkins magic to pack and publish the symbols
- log uploader magic to upload minidumps

/cc @wsnipex @Memphiz @Montellese

### CODE REVIEW

**Comment by koying** [comment]
jenkins build this please

File: configure.ac [hunk 1]
```diff
@@ -2296,6 +2304,21 @@ else
...
 fi
 
+# Google Breakpad
+if test "$configure_breakpad" = "yes"; then
+  AC_MSG_NOTICE($breakpad_enabled)
+  final_message="$final_message\n  Google Breakpad Configured:\tYes"
+  USE_BREAKPAD=1
+  PKG_CHECK_MODULES([BREAKPAD],    [breakpad-client],
+    [INCLUDES="$INCLUDES $BREAKPAD_CFLAGS"; LIBS="$LIBS $BREAKPAD_LIBS"],
+    AC_MSG_ERROR($missing_library))
```
Comment (wsnipex) [inline configure.ac]: please:
AC_MSG_ERROR("breakpad-client not found").

and you could move the enabled notice and final_message down after the ac_define.

File: tools/depends/native/google-breakpad-native/Makefile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+# configuration settings
+CONFIGURE= ./configure --prefix=$(PREFIX)
+
+CLEAN_FILES=$(ARCHIVE) $(PLATFORM)
+
+all: .installed-$(PLATFORM)
+
+$(TARBALLS_LOCATION)/$(ARCHIVE):
+	cd $(TARBALLS_LOCATION); $(RETRIEVE_TOOL) -Ls -f "http://www.semperpax.be/owncloud/public.php?service=shorty_relay&id=7sq73397dqY" -o $(TARBALLS_LOCATION)/$(ARCHIVE)
```
Comment (wsnipex) [inline tools/depends/native/google-breakpad-native/Makefile]: move to mirrors before merge please ;)

File: tools/depends/native/google-breakpad-native/Makefile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	#cd $(TARBALLS_LOCATION); $(RETRIEVE_TOOL) $(RETRIEVE_TOOL_FLAGS) $(BASE_URL)/$(ARCHIVE)
+
+$(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE) $(DEPS)
+	rm -rf $(PLATFORM)/*; mkdir -p $(PLATFORM)
+	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE)
+	cd $(PLATFORM); $(CONFIGURE)
+
+$(LIBDYLIB): $(PLATFORM)
+	make -C $(PLATFORM)
```
Comment (wsnipex) [inline tools/depends/native/google-breakpad-native/Makefile]: $(MAKE) please

File: tools/depends/target/google-breakpad/Makefile [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	cd $(TARBALLS_LOCATION); $(RETRIEVE_TOOL) -Ls -f "http://www.semperpax.be/owncloud/public.php?service=shorty_relay&id=7sq73397dqY" -o $(TARBALLS_LOCATION)/$(ARCHIVE)
+	#cd $(TARBALLS_LOCATION); $(RETRIEVE_TOOL) $(RETRIEVE_TOOL_FLAGS) $(BASE_URL)/$(ARCHIVE)
+
+$(PLATFORM): $(TARBALLS_LOCATION)/$(ARCHIVE) $(DEPS)
+	rm -rf $(PLATFORM)/*; mkdir -p $(PLATFORM)
+	cd $(PLATFORM); $(ARCHIVE_TOOL) $(ARCHIVE_TOOL_FLAGS) $(TARBALLS_LOCATION)/$(ARCHIVE)
+	cd $(PLATFORM); $(CONFIGURE)
+
+$(LIBDYLIB): $(PLATFORM)
+	make -C $(PLATFORM)
```
Comment (wsnipex) [inline tools/depends/target/google-breakpad/Makefile]: $(MAKE)

File: tools/depends/native/google-breakpad-native/Makefile (hunk 2 above)
Comment (koying) [inline tools/depends/native/google-breakpad-native/Makefile]: Chicken and egg ;)

Comment (wsnipex) [inline tools/depends/native/google-breakpad-native/Makefile]: uploaded to mirrors

**Comment by koying** [comment]
jenkins build this please

**Comment by koying** [comment]
jenkins build and merge

**Pull request merged**